.github/actions/send-teams-notification/compute-message.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash -e
+
+COMPUTED_MESSAGE=""
+
+if [ -n "$NEEDS" ]; then
+  COMPUTED_MESSAGE="$NEEDS\n\n"
+fi
+
+if [[ -n "$BLOCK_MESSAGE" && "$APPEND" == 'true' ]] || [ -z "$BLOCK_MESSAGE" ]; then
+  case $EVENT_NAME in
+    pull_request)
+    COMPUTED_MESSAGE="${COMPUTED_MESSAGE}$PR_TITLE"
+    ;;
+    issues)
+    COMPUTED_MESSAGE="${COMPUTED_MESSAGE}$ISSUE_BODY"
+    ;;
+    *)
+    COMPUTED_MESSAGE="${COMPUTED_MESSAGE}$COMMIT_MESSAGE"
+    ;;
+  esac
+fi
+
+if [ -n "$BLOCK_MESSAGE" ]; then
+  if [[ "$APPEND" == 'true' && -n "$COMPUTED_MESSAGE" ]]; then
+    COMPUTED_MESSAGE="${COMPUTED_MESSAGE}\n\n"
+  fi
+  COMPUTED_MESSAGE="${COMPUTED_MESSAGE}$BLOCK_MESSAGE"
+fi
+
+if [ -n "$COMPUTED_MESSAGE" ]; then
+  COMPUTED_MESSAGE="${COMPUTED_MESSAGE}"
+  COMPUTED_MESSAGE=$(printf "${COMPUTED_MESSAGE}" | sed -z 's/\n/\\n/g' | sed -r 's/"/\\\"/g' | sed -e 's/\r//g')
+  # avoid error if message is too long (total message must be less than 3001 characters)
```

Comment:
Actually this particular number was added when some slack limits were hit: we can definitely up this limit but it does feel like it's enough content (especially since the long content usually comes from long commit messages, so we can always get the full info even if it's not in the notif)